### PR TITLE
create a feature resource and guideline

### PR DESCRIPTION
**Pull request by kanuj365**
#2764
create a feature for resource and guideline that can have different types of categories.

### Description
- Create a resource and guideline view page that have resource and category model.
- In view page have 3 types of data like category, mark as read , user's suggestions.
- Create a resource and guideline view page for users that user can see all types of resource's category.
- On user's resource and guideline view page that have 3 type of table data like category, mark as read and post_suggestion.  
 

### Checklist:

- [x] I have performed a self-review of my own code.

### CODE REVIEW

**Comment by kanuj365** [comment]
image

File: Modules/HR/Http/Controllers/EmployeeController.php [hunk 1]
```diff
@@ -94,4 +96,27 @@ public function employeeWorkHistory(Employee $employee)
...
+                 'user_id' => isset($data['mark_as_read'][$index]) && $data['mark_as_read'][$index] ? $employee->id : null,
+                'mark_as_read' => isset($data['mark_as_read'][$index]) ? $data['mark_as_read'][$index] : null],
+                [
+                'post_suggestions' => isset($data['post_suggestion'][$index]) && isset($data['mark_as_read'][$index]) ? $data['post_suggestion'][$index] : null
+                ]
+            );
+        }
+
+        return redirect()->back()->with('success', 'Resources saved successfully');
+    }
```
Comment (P4NK4J) [inline Modules/HR/Http/Controllers/EmployeeController.php:115]: Controller should not be responsible for executing database queries, please put the create logic in service. Also, this function name doesn't make sense what its doing, please check

File: Modules/HR/Http/Controllers/EmployeeController.php [hunk 2]
```diff
@@ -94,4 +96,27 @@ public function employeeWorkHistory(Employee $employee)
...
+    public function resources(Employee $employee)
```
Comment (P4NK4J) [inline Modules/HR/Http/Controllers/EmployeeController.php:116]: Can we rename this function to have getAllResources kind of name ?

File: Modules/HR/Resources/views/guidelines-resources/index.blade.php [hunk 3]
```diff
@@ -1,19 +1,73 @@
...
+                                @csrf
+                                <a href="#" data-toggle="modal" data-target="#edit-Modal" id="resource_id" resource_id={{ $resource->id }} action_url={{route('resources.update', $resource->id)}} resource_link="{{$resource->resource_link}}" category_id="{{$resource->category->id}}" class="pr-1 btn btn-link rg_edit_btn"><i class="text-success fa fa-edit fa-lg"></i></a>
+                                <button type="submit" id="resource_id" data-toggle="modal" class="pl-1 btn btn-link"><i class="text-danger fa fa-trash fa-lg"></i></button>
+                            </form>
+                        </div>
+                    </div>
+                </td>
+                <td class="text-center">
+                    @foreach ($userSuggestions as $userSuggestion)
+                    @if (!$userSuggestion->mark_as_read && $userSuggestion->category==$resource->category->name)
```
Comment (P4NK4J) [inline Modules/HR/Resources/views/guidelines-resources/index.blade.php:51]: check indentation and formatting, don't push unhygienic code

**Review by P4NK4J — CHANGES_REQUESTED** [review]
left some comments,

File: Modules/HR/Resources/views/guidelines-resources/index.blade.php [hunk 4]
```diff
@@ -1,19 +1,73 @@
...
         </div>
     </div>
-    @endforeach
-</div>
+    @include('hr::guidelines-resources.hr-category-modal')
+    @include('hr::guidelines-resources.hr-resources-modal')
 </div>
-@endsection
\ No newline at end of file
+@endsection
```
Comment (kanuj365) [inline Modules/HR/Resources/views/guidelines-resources/index.blade.php:73]: In blade file i used php intelligence for HTML elements formatting for every line.

Comment (kanuj365) [inline Modules/HR/Resources/views/guidelines-resources/index.blade.php:73]: I always used PHP Intelsense for HTML element format

**Review by P4NK4J — CHANGES_REQUESTED** [review]
Resource link not appearing correctly
image

If there is no reader, the message should not say please click on avatar
Screenshot 2023-01-21 at 3 49 11 PM

The size of buttons is very large, can we make the buttons text in a single line ?

Deleting a resource is displaying the message, `category deleted successfully`, it should say `resource deleted successfully`, because we are not deleting the category

Upon saving the suggestion, I am getting empty values on the dashboard. The checkbox should be checked if I have marked it as read, and the suggestion should be stored, in case I edit the suggestion, it should be updated on HR resources page as well. Also, the tab should read `Resources` here. 
Also, there should be functionality to mark as unread. The table header can say `Action`, and the UI can be like a checkbox and mark as read/unread message next to it
image

Let's make suggestions color black, because blue is used for denoting a hyperlink. 
image

**Comment by kanuj365** [comment]
image

**Comment by kanuj365** [comment]
image
image

**Pull request closed without merging**